near/omni-prover/wormhole-omni-prover-proxy/src/parsed_vaa.rs

```diff
@@ -140,9 +143,8 @@ impl TryInto<InitTransferMessage> for ParsedVAA {
     type Error = String;
 
     fn try_into(self) -> Result<InitTransferMessage, String> {
-        let data: &[u8] = &self.payload[1..];
-        let transfer = InitTransferWh::abi_decode(data, true).map_err(stringify)?;
-
+        let transfer =
```

Comment:
Should we also check here messageType?

Reply:
No, since `InitTransferWh` and `FinTransferWh` have different fields it simply won't be able to decode encoded `FinTransfer` as `InitTransferWh` or vice versa

https://github.com/Near-One/omni-bridge/blob/071ed8ec1c6e969af3ef6bbcdee518b5d49b2524/near/omni-prover/wormhole-omni-prover-proxy/src/parsed_vaa.rs#L115-L133

Reply:
it is already verified here
https://github.com/Near-One/omni-bridge/blob/071ed8ec1c6e969af3ef6bbcdee518b5d49b2524/near/omni-prover/wormhole-omni-prover-proxy/src/lib.rs#L74

Reply:
Yes, I mean this check. To me, it seems logical to check this byte inside try_into.